executor/builder.go

```diff
@@ -789,53 +762,52 @@ func (b *executorBuilder) buildTopN(v *plan.PhysicalTopN) Executor {
 }
 
 func (b *executorBuilder) buildNestedLoopJoin(v *plan.PhysicalHashJoin) *NestedLoopJoinExec {
+	leftChild := b.build(v.Children()[0])
+	if b.err != nil {
+		b.err = errors.Trace(b.err)
+		return nil
+	}
+	rightChild := b.build(v.Children()[1])
+	if b.err != nil {
+		b.err = errors.Trace(b.err)
+		return nil
+	}
+	joinSchema := expression.MergeSchema(leftChild.Schema(), rightChild.Schema())
 	for _, cond := range v.EqualConditions {
-		cond.GetArgs()[0].(*expression.Column).ResolveIndices(v.Schema())
```

Comment:
use v.Schema() is not correct?

Reply:
When NestedLoop is used for semi join, the equal condition will fail to resolve, because the v.Schema only contains the big table.